backend/box/box.go

```diff
@@ -966,6 +965,15 @@ func (f *Fs) Copy(ctx context.Context, src fs.Object, remote string) (fs.Object,
 		return nil, err
 	}
 
+	// check if dest already exists
+	item, err := f.preUploadCheck(ctx, leaf, directoryID, src.Size())
+	if err != nil {
+		return nil, err
+	}
+	if item != nil { // dest already exists, fall back to manual copy
+		return nil, fs.ErrorCantCopy
```

Comment:
Could we just delete the existing file here?

Or if paranoid
- rename existing file with random suffix
- do copy
- if successful delete existing file
- if unsucessful rename exsting file back

The success/unsuccess could be done in a `defer` function which looks at the result of `err`.

Reply:
I like the paranoid option. I suppose there's still a slight risk of leaving things in a messy state if the rename-back fails, but I don't know if that's enough of a concern to worry about.

Reply:
We do the paranoid option in one of the other backends I think.

Perhaps less failure prone would be
- do new copy to random suffix
- if successful delete existing then rename copy
    - or maybe just rename the copy over the existing if that is supported
- if unsucessful probably nothing to do